Add reference policy contract

Melvillian opened the pull request.

Melvillian commented:
@fnerdman there is an important decision that still needs to be decided with this contract. If you compare my implementation to [what the spec says about Policy's](https://github.com/flashbots/rollup-boost/blob/flashtestations/specs/flashtestations.md#policy-operations) you'll see that they don't match. The reason is that while the spec's functions:

```
// Check if an address is allowed under any workload in a policy
function isAllowedPolicy(policyId, teeAddress) → boolean

// Governance operations
function addWorkloadToPolicy(policyId, workloadId)
function removeWorkloadFromPolicy(policyId, workloadId)
```

all take a `policyId` as an argument, my implementation does not.The spec implies there will be a singleton PolicyRegistry contract that you pass `policyIds` to. But from our DM conversations, we talked about each policy being its own contract that anyone is allowed to implement and deploy for their own purposes.

So we either need to go with my implementation and update the spec, or discuss the tradeoffs of a PolicyRegistry and I'll go implement that here.

I like the flexibility of not having a PolicyRegistry, but one benefit of a PolicyRegistry is in the UX of policies; because there is a singleton contract holding all the policies, it'll be easier for us to build tools that let users explore and select those policies. idk if this is a feature we want, but its easier with the PolicyRegistry. 

I overall prefer _not_ going with the PolicyRegistry and keeping it flexible, but I'd like to hear your thoughts on this.

fnerdman reviewed (changes requested):
still missing some test cases for the policy ref impl. LGTM otherwise

Melvillian commented:
See my reply to [your comment about verifying more than just workloadIds](https://github.com/flashbots/flashtestations/pull/11#discussion_r2120692599).

I also added tests.

I also fixed a critical bug in `isValidWorkload` where we weren't checking the `isValid` field.
